CI/Dockerfile

```diff
@@ -34,63 +57,147 @@ RUN apt-get -y update; \
-RUN /root/etc/CI/docker/build_geant4.sh
-
+RUN mkdir -p ${geant4_build_dir}/build && \
+    cd ${geant4_build_dir} && \
+    wget https://gitlab.cern.ch/geant4/geant4/-/archive/v${geant4_version}/${geant4_tarball} --no-check-certificate && \
+    tar -xzf ${geant4_tarball} && \
+    cd build && \
+    cmake ../${geant4_basename} -DCMAKE_INSTALL_RPATH=${geant4_install_dir}/lib \
+                                -DCMAKE_INSTALL_PREFIX=${geant4_install_dir} \
+                                -DGEANT4_USE_SYSTEM_EXPAT=OFF \
+                                -DCMAKE_BUILD_TYPE=Release \
+                                -DCMAKE_CXX_COMPILER=${CXX} \
+                                -DCMAKE_C_COMPILER=${CC} \
+                                -DBUILD_STATIC_LIBS=ON && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${geant4_build_dir}
+
+# these two double down ENVs don't appear to be used yet
+# However they were in orginal dockerfile, so they have been kept
+# TODO add double down to the dockerfile
 ENV double_down_build_dir=${build_dir}/double-down/
 ENV double_down_install_dir=${install_dir}/double-down/
 
-# Build Embree
-#TODO move sh file contents into this Dockerfile
-RUN /root/etc/CI/docker/build_embree.sh
+ENV embree_install_dir=${install_dir}/embree
+ENV embree_build_dir=${build_dir}/embree
+
+RUN mkdir -p ${embree_build_dir}/build && \
+    cd ${embree_build_dir} && \
+    git clone -b ${EMBREE_BRANCH} https://github.com/embree/embree && \
+    cd build && \
+    cmake ../embree -DCMAKE_INSTALL_PREFIX=${embree_install_dir} \
+                    -DEMBREE_TASKING_SYSTEM=INTERNAL \
+                    -DEMBREE_ISPC_SUPPORT=OFF \
+                    -DEMBREE_TUTORIALS=OFF \
+                    -DEMBREE_TBB_ROOT=/usr && \
+    make -j${ci_jobs} && \
+    make -j${ci_jobs} install && \
+    cd && \
+    rm -rf ${embree_build_dir}
+
 
 FROM external_deps AS hdf5
 
-# Set HDF5 env variable
+# accessing gloabl ARGs in build stage
+ARG HDF5_VERSION_major
+ARG HDF5_VERSION
+ARG install_dir
+ARG build_dir
+ARG ci_jobs
+ARG CXX
+ARG CC
+
 ENV hdf5_build_dir=${build_dir}/hdf5
 ENV hdf5_install_dir=${install_dir}/hdf5
 
-# Build HDF5
-# HDF5 argument possible value: 1.10.4 or system
-ARG HDF5=1.10.4
-ENV HDF5_VERSION=${HDF5}
-#TODO move sh file contents into this Dockerfile
-RUN /root/etc/CI/docker/build_hdf5.sh
+RUN mkdir -p ${hdf5_build_dir}/build && \
+    cd ${hdf5_build_dir} && \
+    wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-${HDF5_VERSION_major}/hdf5-${HDF5_VERSION}/src/hdf5-${HDF5_VERSION}.tar.gz && \
+    tar -xzf hdf5-${HDF5_VERSION}.tar.gz && \
+    cd build && \
+    ../hdf5-${HDF5_VERSION}/configure --enable-shared \
+        --prefix=${hdf5_install_dir} \
+        CXX=${CXX} \
+        CC=${CC} && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${hdf5_build_dir}
+
 
 FROM hdf5 AS moab
 
-ENV PYTHONPATH=/root/build_dir/moab/bld/pymoab/lib/python3.6/site-packages
```

Comment:
I think the CI is failing because this line is not present anymore :)

Reply:
Yes I am sure you are right about this @bam241 I just wanted to avoid putting in this ENV to get things working. Ideally the moab install would sort out this

Reply:
Can't we prepend this to the `make` command (or `make install`) for MOAB below rather than embed it in the `ENV`?